lib/src/code/code.dart

```diff
@@ -62,7 +65,12 @@ class Code {
     } else {
       final parser = HighlightFoldableBlockParser();
```

Comment:
`Code` should not fiddle with the languages.
There should be a global factory that returns a parser for `Mode` (or even just a map, see `SingleLineComments` class). This factory should be called here.

This means that `AbstractFoldableBlockParser` should have an abstract `parse` method and define its arguments, so that all subclasses should follow that signature even if they do not use some of the arguments.

This also means that the Python parser should extend `AbstractFoldableBlockParser`.

Reply:
awesome solution. Thanks!